rmw_fastrtps_cpp/src/rmw_service.cpp

```diff
@@ -237,7 +241,7 @@ rmw_create_service(
     delete info;
   }
 
-  if (rmw_service->service_name != nullptr) {
+  if (rmw_service && rmw_service->service_name != nullptr) {
```

Comment:
it looks to me like a redundant test, because `rmw_service` was checked for null in line 202. Also, try to use the same notation of `(!rmw_service->service_name)` instead of `!= nullptr`

Reply:
Actually, if it goes to `fail` before `rmw_service` is really initialized (a. k. a assigned to a valid address at line 201), even though the `rmw_service` is `nullptr`, but `rmw_service->service_name` is actually not **0x0** and it should be some value like **0x10**, then there will be "Segmentation fault (core dumped)" error

Reply:
please apologize, I haven't noticed the modified condition was inside the `fail`. Your change is completely correct here.